localstack/testing/scenario/provisioning.py

```diff
@@ -16,85 +23,168 @@
 
 LOG = logging.getLogger(__name__)
 CDK_BOOTSTRAP_PARAM = "/cdk-bootstrap/hnb659fds/version"
-WAITER_CONFIG_AWS = {"Delay": 10, "MaxAttempts": 1000}
-WAITER_CONFIG_LS = {"Delay": 1, "MaxAttempts": 500}
+WAITER_CONFIG_AWS = {
+    "Delay": 6,
+    "MaxAttempts": 600,
+}  # total timeout ~1 hour (6 * 600 = 3_600 seconds)
+WAITER_CONFIG_LS = {"Delay": 1, "MaxAttempts": 600}  # total timeout ~10 minutes
+TEMPLATE_DIR_NAME = "cdk_templates"
 
 
-def cleanup_s3_bucket(s3_client: "S3Client", bucket_name: str, delete_bucket: bool = False):
-    LOG.debug(f"Cleaning provisioned S3 Bucket {bucket_name}")
+def _cleanup_s3_bucket(s3_client: "S3Client", bucket_name: str, delete_bucket: bool = False):
```

Comment:
this function is currently also used in some of the scenario tests to cleanup buckets that have been created manual e.g the corresponding tear down for add_custom_setup_provisioning_step.

I think it makes more sense to move this function to a cdk-utility?

Reply:
It also has a lot in common with the S3 cleaning fixture. I had an issue with the above, as the bucket created with CDK was versioned so this didn't clean up properly. Maybe the code inside the fixture could be extracted as an util in S3, the used in both the fixture and this cdk util?

Reply:
Left it in there for now but fixed the naming for the usage in tests again. We can tackle this in a follow-up :+1: